zookeeper/cli.py

```diff
@@ -1,185 +1,63 @@
+        except:
+            self.fail(
+                "configuration parameters must be of the form 'key=value', where "
+                "the key contains only alpha-numeric characters, underscores, and "
+                f"full-stops, but received '{str_value}'.",
+                param,
+                ctx,
             )
-            build_model = registry.get_model_function(model_name)
-            hparams = registry.get_hparams(model_name, hparams_set)
-            if hparams_str:
-                hparams.parse(hparams_str)
-            click.echo(hparams)
-
-            if output_dir is None:
-                time_stamp = datetime.now().strftime("%Y%m%d_%H%M")
-                output_dir = os.path.join(
-                    output_prefix,
-                    dataset_name,
-                    model_name,
-                    f"{hparams_set}_{time_stamp}",
-                )
-
-            function(build_model, dataset, hparams, output_dir, **kwargs)
-
-        return train
 
-    return decorator
-
-
-@cli.command()
-@click.argument("datasets", type=str, nargs=-1)
-@click.option("--data-dir", type=str, help="Directory with training data.")
-@click.option("--progressbar", default=False, help="Show tfds progressbar")
-def prepare(datasets, data_dir, progressbar):
-    """Downloads and prepares datasets for reading."""
-    import tensorflow_datasets as tfds
-
-    if not progressbar:
-        tfds.disable_progress_bar()
+        try:
+            value = parse_value_from_string(value)
+        except:
+            self.fail(
+                f"unable to parse value of configuration parameter {str_value}. The "
+                "only supported types are `int`, `float`, `str`, `None`, and "
+                "lists/tuples of the above.",
+                param,
+                ctx,
+            )
 
-    for dataset in datasets:
-        tfds.builder(dataset, data_dir=data_dir).download_and_prepare()
-        click.secho(f"Finished preparing dataset: {dataset}", fg="green")
+        return key, value
 
 
-@cli.command(context_settings=dict(allow_extra_args=True, ignore_unknown_options=True))
-@click.argument("model", required=False)
-@click.option("--dataset", help="Tensorflow dataset name.")
-@click.option(
-    "--output-prefix",
-    default=os.path.expanduser("~/zookeeper-logs"),
-    help="Directory prefix used to save model checkpoints and logs.",
-)
-@click.option("--output-dir", "--logdir", help="Directory containing checkpoints.")
-def tensorboard(model, dataset, output_prefix, output_dir):
-    """Start TensorBoard to monitor model training."""
-    if output_dir is None:
-        output_dir = output_prefix
-        if dataset:
-            output_dir = os.path.join(output_dir, dataset)
-            if model:
-                output_dir = os.path.join(output_dir, model)
-    click.secho(f"Starting TensorBoard at: {output_dir}", fg="blue")
-    os.system(f"tensorboard --logdir={output_dir}")
+def add_job_to_cli(job_cls: type):
```

Comment:
I think a decorator is a perfectly fine API to have (and I like decorators). Since we are already overwriting `__init_subclass__` on many of our classes, should we add a `CLIJob` instead?

Reply:
I think the problem with doing this is that you might want a base class to not be runnable, but certain specific subclasses to be runnable. We could get around that by having those specific subclasses inherit from both `Parent` and `CLIJob`. What do you think about that?

Reply:
I don't see an immediate use case for that, but I think the decorator makes more sense then, since we are just annotating something to be runnable via CLI.